index.js

```diff
@@ -418,43 +419,62 @@ function createPanZoom(domElement, options) {
     zoomByRatio(clientX, clientY, ratio);
   }
 
-  function centerOn(ui) {
-    var parent = ui.ownerSVGElement;
-    if (!parent)
-      throw new Error('ui element is required to be within the scene');
-
+  function centerOn(smooth) {
     // TODO: should i use controller's screen CTM?
-    var clientRect = ui.getBoundingClientRect();
+    var clientRect = owner.getBoundingClientRect();
     var cx = clientRect.left + clientRect.width / 2;
     var cy = clientRect.top + clientRect.height / 2;
 
+    var parent = owner.parentElement.parentElement;
     var container = parent.getBoundingClientRect();
     var dx = container.width / 2 - cx;
     var dy = container.height / 2 - cy;
 
-    internalMoveBy(dx, dy, true);
+    internalMoveBy(dx, dy, smooth);
   }
 
-  function internalMoveBy(dx, dy, smooth) {
-    if (!smooth) {
-      return moveBy(dx, dy);
-    }
+  function zoomToFit(smooth) {
+    // TODO: should i use controller's screen CTM?
+    var clientRect = owner.getBoundingClientRect();
+    var cx = clientRect.left + clientRect.width/2;
+    var cy = clientRect.top + clientRect.height/2;
 
-    if (moveByAnimation) moveByAnimation.cancel();
+    var parent = owner.parentElement.parentElement;
```

Comment:
I'm not following this. Why are we fetching `parentElement.parentElement`? Any way to avoid it?

Same applies to the use in the `centerOn()` method